packages/shared/src/hooks/post/useLeanPostActions.ts

```diff
@@ -0,0 +1,464 @@
+import { useCallback, useContext } from 'react';
+import { QueryClient, useMutation, useQueryClient } from 'react-query';
+import {
+  banPost,
+  deletePost,
+  demotePost,
+  Post,
+  promotePost,
+  updatePinnedPost,
+  UserPostVote,
+  VOTE_POST_MUTATION,
+} from '../../graphql/posts';
+import {
+  ToggleVoteProps,
+  upvoteMutationKey,
+  voteMutationHandlers,
+} from '../vote';
+import { postAnalyticsEvent } from '../../lib/feed';
+import { Origin } from '../../lib/analytics';
+import useReportPost from '../useReportPost';
+import { useAnalyticsContext } from '../../contexts/AnalyticsContext';
+import { AuthTriggers } from '../../lib/auth';
+import { graphqlUrl } from '../../lib/config';
+import { useRequestProtocol } from '../useRequestProtocol';
+import AuthContext from '../../contexts/AuthContext';
+import useBookmarkPost from '../useBookmarkPost';
+import { ToastSubject, useToastNotification } from '../useToastNotification';
+import { PromptOptions, usePrompt } from '../usePrompt';
+import { Roles } from '../../lib/user';
+import { SourcePermissions, SourceType } from '../../graphql/sources';
+import { SourceActionArguments, TagActionArguments } from '../useTagAndSource';
+import useMutateFilters from '../useMutateFilters';
+import AlertContext from '../../contexts/AlertContext';
+import useFeedSettings from '../useFeedSettings';
+
+const multiPostTransformer = (
+  posts: Post[],
+  id: string,
+  update: (oldPost: Post) => Partial<Post>,
+): Post[] =>
+  posts.map((post) =>
+    post.id === id
+      ? {
+          ...post,
+          ...update(post),
+        }
+      : post,
+  );
+
+const updateMap = {
```

Comment:
The final idea is to have one cache for post states, regardless of who needs it. 
As the "intermediate" solution I thought we could have this mapping per query to say what transformer to run on the provided cache.  (one issue still is how to update multiple places)

Reply:
I suggest looking for normalized cache solutions or ignoring it, and in any way I wouldn't recommend doing it in the same PR. Developing one ourselves is too much effort and bug-prone